### PR TITLE
[WIP][ResourceBundle] Create integration tests

### PR DESCRIPTION
**Pull request by tuka217**
Q 	|A
-----|-----
Bug fix? |	no
New feature?| 	no
BC breaks? 	|no
Deprecations? 	|no
Fixed tickets 	|-
License 	|MIT
Doc PR 	|-

### CODE REVIEW

File: src/Sylius/Bundle/ResourceBundle/composer.json [hunk 1]
```diff
@@ -35,10 +35,13 @@
         "symfony/twig-bundle": "^2.7",
         "white-october/pagerfanta-bundle": "^1.0",
         "willdurand/hateoas-bundle": "^0.4|^1.0"
+
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/composer.json]: Redundant blank line

File: src/Sylius/Bundle/ResourceBundle/composer.json [hunk 2]
```diff
@@ -59,7 +62,10 @@
         "bin-dir": "bin"
     },
     "autoload": {
-        "psr-4": { "Sylius\\Bundle\\ResourceBundle\\": "" }
+        "psr-4": {
+            "Sylius\\Bundle\\ResourceBundle\\": "",
+            "AppBundle\\": "tests/src/AppBundle/"
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/composer.json]: This should go to `autoload-dev`

File: src/Sylius/Bundle/ResourceBundle/phpunit.xml.dist [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+         backupGlobals="false"
+         colors="true"
+         bootstrap="tests/app/autoload.php"
+>
+    <testsuites>
+        <testsuite name="SyliusResourceBundle Test Suite">
+            <directory>./tests/</directory>
+        </testsuite>
+    </testsuites>
+</phpunit>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/phpunit.xml.dist]: Missing blank line at the end of file

File: src/Sylius/Bundle/ResourceBundle/tests/app/autoload.php [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+$loader = require __DIR__.'/../../vendor/autoload.php';
+
+require __DIR__.'/AppKernel.php';
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/tests/app/autoload.php]: Missing blank line

File: src/Sylius/Bundle/ResourceBundle/tests/app/AppKernel.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class AppKernel extends Kernel
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function registerBundles()
+    {
+        return array(
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/tests/app/AppKernel.php]: Remember about short syntax

File: src/Sylius/Bundle/ResourceBundle/tests/app/config.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
+framework:
+    secret: "Three can keep a secret, if two of them are dead."
+    router:
+        resource: %kernel.root_dir%/routing.yml
+    session: ~
+    form: ~
+    templating:     { engines: ['twig'] }
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/tests/app/config.yml]: Too many spaces

File: src/Sylius/Bundle/ResourceBundle/tests/app/routing.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+app_book_api:
+    resource: |
+        alias: app.book
+    type: sylius.resource
+
+sylius_resource_controller:
+    resource: |
+        alias: sylius.resource_controller
+    type: sylius.resource
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/tests/app/routing.yml]: Missing blank line

File: src/Sylius/Bundle/ResourceBundle/tests/src/AppBundle/Entity/Book.php [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class Book
+{
+    /**
+     * @var string
+     */
+    private $title;
+
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/tests/src/AppBundle/Entity/Book.php]: Redundant blank line

File: src/Sylius/Bundle/ResourceBundle/tests/src/AppBundle.php [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpKernel\Bundle\Bundle;
+
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class AppBundle extends Bundle
+{
+
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/tests/src/AppBundle.php]: Redundant blank line

File: src/Sylius/Bundle/ResourceBundle/tests/app/routing.yml [hunk 10]
```diff
@@ -0,0 +1,9 @@
+app_book_api:
+    resource: |
+        alias: app.book
+    type: sylius.resource
+
+sylius_resource_controller:
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ResourceBundle/tests/app/routing.yml]: Why do you register crud for `sylius_resource_controller`?

File: src/Sylius/Bundle/ResourceBundle/.gitignore [hunk 11]
```diff
@@ -3,3 +3,6 @@ bin/
 
 composer.phar
 composer.lock
+
+tests/app/cache
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/.gitignore]: Why don't we use a temporary directory just like in `ThemeBundle/Tests/Functional/WebTestCase::getTmpDirPath`?

File: src/Sylius/Bundle/ResourceBundle/tests/app/AppKernel.php [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+            new Symfony\Bundle\FrameworkBundle\FrameworkBundle(),
+            new FOS\RestBundle\FOSRestBundle(),
+            new JMS\SerializerBundle\JMSSerializerBundle($this),
+            new Sylius\Bundle\ResourceBundle\SyliusResourceBundle(),
+            new WhiteOctober\PagerfantaBundle\WhiteOctoberPagerfantaBundle(),
+            new Bazinga\Bundle\HateoasBundle\BazingaHateoasBundle(),
+            new Doctrine\Bundle\DoctrineBundle\DoctrineBundle(),
+            new \Symfony\Bundle\TwigBundle\TwigBundle(),
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/tests/app/AppKernel.php]: No `\` needed at the beginning.

File: src/Sylius/Bundle/ResourceBundle/tests/src/AppBundle.php [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ResourceBundle\AppBundle;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpKernel\Bundle\Bundle;
+
+
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/tests/src/AppBundle.php]: Doubled blank line

File: src/Sylius/Bundle/ResourceBundle/tests/src/AppBundle/Entity/Book.php [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+    /**
+     * @return string
+     */
+    public function getTitle()
+    {
+        return $this->title;
+    }
+
+    /**
+     * @param $title
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/tests/src/AppBundle/Entity/Book.php]: Missing typehint

File: src/Sylius/Bundle/ResourceBundle/tests/src/AppBundle/Form/Type/BookType.php [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+
+use Sylius\Bundle\ResourceBundle\Form\Type\AbstractResourceType;
+use Symfony\Component\Form\FormBuilderInterface;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class BookType extends AbstractResourceType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options)
```
Comment (pamil) [inline src/Sylius/Bundle/ResourceBundle/tests/src/AppBundle/Form/Type/BookType.php]: Missing inhertidocs

**Comment by pamil** [comment]
Why don't we use https://github.com/matthiasnoback/SymfonyDependencyInjectionTest for it?

File: src/Sylius/Bundle/ResourceBundle/.gitignore (hunk 11 above)
Comment (tuka217) [inline src/Sylius/Bundle/ResourceBundle/.gitignore]: We don't use this method in ThemeBundle. Are you sure that you saw this in .gitignore file?

File: src/Sylius/Bundle/ResourceBundle/test/app/AppKernel.php [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+            new FOS\RestBundle\FOSRestBundle(),
+            new JMS\SerializerBundle\JMSSerializerBundle(),
+            new Sylius\Bundle\ResourceBundle\SyliusResourceBundle(),
+            new WhiteOctober\PagerfantaBundle\WhiteOctoberPagerfantaBundle(),
+            new Bazinga\Bundle\HateoasBundle\BazingaHateoasBundle(),
+            new Doctrine\Bundle\DoctrineBundle\DoctrineBundle(),
+            new Symfony\Bundle\TwigBundle\TwigBundle(),
+            new AppBundle\AppBundle(),
+            new Sensio\Bundle\GeneratorBundle\SensioGeneratorBundle(),
+            new Symfony\Bundle\MonologBundle\MonologBundle(),
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/ResourceBundle/test/app/AppKernel.php]: We do not need Monolog?

File: src/Sylius/Bundle/ResourceBundle/test/app/AppKernel.php [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+            new Sensio\Bundle\DistributionBundle\SensioDistributionBundle(),
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/ResourceBundle/test/app/AppKernel.php]: Do we need this bundle?

File: src/Sylius/Bundle/ResourceBundle/test/app/config/config.yml [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+framework:
+    #esi:             ~
+    #translator:      { fallback: "%locale%" }
+    secret:          "%secret%"
+    router:
+        resource: "%kernel.root_dir%/config/routing.yml"
+        strict_requirements: ~
+    form:            ~
+    csrf_protection: ~
+    validation:      { enable_annotations: true }
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/ResourceBundle/test/app/config/config.yml]: Not needed?

File: src/Sylius/Bundle/ResourceBundle/test/app/config/parameters.yml [hunk 19]
```diff
@@ -0,0 +1,17 @@
...
+    database_password: Whoa_68!
+    mailer_transport: smtp
+    mailer_host: 127.0.0.1
+    mailer_user: null
+    mailer_password: null
+    locale: en
+    secret: ThisTokenIsNotSoSecretChangeIt
+    debug_toolbar: true
+    debug_redirects: false
+    use_assetic_controller: true
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/ResourceBundle/test/app/config/parameters.yml]: Not needed?

File: src/Sylius/Bundle/ResourceBundle/test/app/AppKernel.php [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+            new FOS\RestBundle\FOSRestBundle(),
+            new JMS\SerializerBundle\JMSSerializerBundle(),
+            new Sylius\Bundle\ResourceBundle\SyliusResourceBundle(),
+            new WhiteOctober\PagerfantaBundle\WhiteOctoberPagerfantaBundle(),
+            new Bazinga\Bundle\HateoasBundle\BazingaHateoasBundle(),
+            new Doctrine\Bundle\DoctrineBundle\DoctrineBundle(),
+            new Symfony\Bundle\TwigBundle\TwigBundle(),
+            new Sensio\Bundle\GeneratorBundle\SensioGeneratorBundle(),
+            new AppBundle\AppBundle(),
+
```
Comment (Zales0123) [inline src/Sylius/Bundle/ResourceBundle/test/app/AppKernel.php]: Redundant blank line.

File: src/Sylius/Bundle/ResourceBundle/test/app/AppKernel.php [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+        $loader->load(__DIR__.'/config/config.yml');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getContainerBaseClass()
+    {
+        if ('test' === $this->environment) {
+            return '\PSS\SymfonyMockerContainer\DependencyInjection\MockerContainer';
```
Comment (Zales0123) [inline src/Sylius/Bundle/ResourceBundle/test/app/AppKernel.php]: Can't we use ``MockerContainer::class``?

**Comment by pjedrzejewski** [comment]
Nice work Ania, thank you!

**Pull request merged**